### PR TITLE
INITIALIZATION

### PR DESCRIPTION
**Pull request by stino-x**
Performed basic git-flow and in stalled necessary fundamental dependencies
- Added in foundational components

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        <Route path="form" element={<Form />} />
+      </Route>,
+    ),
+  );
+
+  return (
+    <ContextProvider>
+      <RouterProvider router={router} />
+    </ContextProvider>
+  );
```
Comment (Johnadibe) [inline src/App.js]: > - [ ] I want to acknowledge your dedication to using the Context API for handling state – that's great! Yet, I'd suggest taking a step back and removing the code you've put into setting up the Context API. You'll soon be delving into Redux, which will be our focus in the upcoming week for state management. For now, it's good to practice working with React's useState hooks.👍

**Review by Johnadibe — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️
### Hi @stino-x👋🏽, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

mario_ladders

## Highlights
✅ No linter errors✔️
✅ Created a /components directory for reusable components✔️
✅ Added 2 routes in the application in which each should render a page component 👍🏽
✅ Create components inside of the /components directory✔️


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 1 above)
Comment (stino-x) [reply src/App.js]: Hi, i get your input but in the project requirement there is no set way required for me to manage my state so as i understand i can share and manage my state using anyhow i like within the realms of using react

**Review by Nemwel-Boniface — COMMENTED** [review]
## Hello @stino-x   👋,

I am Nemwel your reviewer.


## _This is an additional review as a result of the second opinion request._

I have received your second opinion request with a complaint about the previous reviewer @Johnadibe  Below are my findings:

While you are right in your comment that you can manage your state in the way that you deem fit in the realm of React, I am afraid that the reviewer is right by requesting you to delete the Context API config because of the following reasons:
1. This being the setup, you are not really going to need to manage any state and that could easily be done by simply passing state from parent to child via props and not really needing any state management tool.
2. You will learn a new tool called Redux which is the preferred way of managing state which you will learn from next week. The reviewer is helping to make your work easier by ensuring that you only implement what is required by the project requirements and avoid any additional feature that might bring your problems in future milestones.

**What you need to do:**

KIndly refer tp the incline comment that was left by @Johnadibe and kindly implement the changes that he requested. When you make the changes, please feel free to request for another reviewer in your dashboard.


In case anything is not 100% clear tag me @Nemwel-Boniface so that I get the notification.

Keep rocking 🚀!

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import Form from './components/Form/Form';
+import Books from './components/Book-List/Books';
+
+function App() {
+  const [BookArray, setBookArray] = useState([]);
+  const [Bookname, setBookName] = useState('');
+  const [Author, setAuthor] = useState('');
+
+  const createBook = (bookname, author) => {
+    const newBook = {
```
Comment (DuaneDave) [inline src/App.js:15]: [OPTIONAL] As a best practice in react, you want to reduce props drilling and avoid unnecessary rerenders, to achieve that, it is advisable to take states that are specific to a component much closer to that component.

**Review by DuaneDave — APPROVED** [review]
Hi @stino-x 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good job utilizing react hooks

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**